src/fsharp/symbols/Exprs.fs

```diff
@@ -959,17 +959,16 @@ module FSharpExprConvert =
             | Const.Int32   i ->  E.Const(box i, tyR)
             | Const.UInt32  i ->  E.Const(box i, tyR)
             | Const.Int64   i ->  E.Const(box i, tyR)
-            | Const.IntPtr   i ->  E.Const(box (nativeint i), tyR)
             | Const.UInt64  i ->  E.Const(box i, tyR)
-            | Const.UIntPtr   i ->  E.Const(box (unativeint i), tyR)
-            | Const.Double   i ->  E.Const(box i, tyR)
-            | Const.Single i ->  E.Const(box i, tyR)
+            | Const.IntPtr  i ->  E.Const(box (nativeint i), tyR)
+            | Const.UIntPtr i ->  E.Const(box (unativeint i), tyR)
+            | Const.Decimal i ->  E.Const(box i, tyR)
```

Comment:
Actually ``Decimal`` should be eliminated by ``TryEliminateDesugaredConstants``.  Did you encounter a case where it wasn't?